setup/src/Magento/Setup/Module/I18n/Parser/Adapter/Html.php

```diff
@@ -30,7 +30,11 @@ class Html extends AbstractAdapter
         "/i18n:\s?'(?<value>[^'\\\\]*(?:\\\\.[^'\\\\]*)*)'/i",
         // <translate args="'System Messages'"/>
         // <span translate="'Examples'"></span>
-        "/translate( args|)=\"'(?<value>[^\"\\\\]*(?:\\\\.[^\"\\\\]*)*)'\"/i"
+        "/translate( args|)=\"'(?<value>[^\"\\\\]*(?:\\\\.[^\"\\\\]*)*)'\"/i",
+        // <a data-bind="attr: { title: $t('This is \' test \' data'), href: '#'} "></a>
+        // <input type="text" data-bind="attr: { placeholder: $t('Placeholder'), title: $t('Title') }" />
+        // Double quotes are not handled correctly in the `attr` binding. Move phrase to the UI component property if needed
+        '/\\$t\(\s*([\'"])(?<value>.*?[^\\\])\1.*?[),]/i'
```

Comment:
Shouldn't we use the same pattern at the one `setup/src/Magento/Setup/Module/I18n/Parser/Adapter/Js.php` instead?

Reply:
I believe for js files it's already covered in https://github.com/magento/magento2/blob/754ea5fcb1ed51281019235b7095d8152a6ffdc6/setup/src/Magento/Setup/Module/I18n/Parser/Adapter/Html.php#L37
But this issue/PR about knockout js inside html

Reply:
Hi @ihor-sviziev, you are right, for JS is already covered. We were discussing it at the public triage meeting, and @sivaschenko highlighted that the regex between JS and HTML should be the same. 

As far as I can see the difference between them is the `?<value>` that is added, and the `/i` that would make the behavior between them different, one would be case sensitive, the other not.

Reply:
@maksymz could you unify the regex between this line and the similar 
 in the JS file?
https://github.com/magento/magento2/blob/754ea5fcb1ed51281019235b7095d8152a6ffdc6/setup/src/Magento/Setup/Module/I18n/Parser/Adapter/Js.php#L32

Reply:
@gabrieldagama @ihor-sviziev 
The whole thing is about the `.html` templates for Knockout components.

The patterns is the same as in the JS file. I've added this part due to the difference in the parser code: `?<value>`.

Though, seems that `\i` modifier is a problem here. `$T()` will not work as function names in JS are case-sensitive. Besides, I've just checked both the previous fix to this file (https://github.com/magento/magento2/issues/11175) and the original pattern. It seems both must be case-sensitive, because:

a) Knockout binding names are case-sensitive. 
The following will not work:
```html
<span data-bind="I18N: 'Foo Bar Baz'"></span>
```

The following works fine:
```html
<span data-bind="i18n: 'Foo Bar Baz'"></span>
```

b) translation via the `translate` attribute works only when the attribute name is lowercase.
The following will not work:
```html
<span TRANSLATE="'Foo Bar Baz'"></span>
```

The following works fine:
```html
<span translate="'Foo Bar Baz'"></span>
```

Current state: translations work only with the lowercase attribute/binding/function call.

Solution 1: remove the `\i` modifier from all three patterns, update unit tests to cover uppercase spelling. We get better translations coverage and the rules will be defined better. No changes to how translations work.

Solution 2: search where `translate` attribute is processed and make it possible to work with uppercase attributes. Though, uppercase tag attributes are probably controversial with the HTML code style.

Think solution 1 is a better one. I'll change the code and update unit tests if agreed.

Regards,
Max

Reply:
Just checked one more way of translating the text. Right now the following does not work as well, the text is shown without translation:
```html
<TRANSLATE args="'Foo Bar Baz'"/>
```

The following works fine:
```html
<translate args="'Foo Bar Baz'"/>
```